server/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -687,49 +689,77 @@ public AnalyzedRelation visitWithQuery(WithQuery node, StatementAnalysisContext
     @Override
     protected AnalyzedRelation visitTable(Table<?> node, StatementAnalysisContext context) {
         QualifiedName tableQualifiedName = node.getName();
+        RelationName relationName = RelationName.of(tableQualifiedName, null);
         SearchPath searchPath = context.sessionSettings().searchPath();
-        AnalyzedRelation relation;
         var relationContext = context.currentRelationContext();
 
         var withQuery = relationContext.parentSources()
-            .getAncestor(RelationName.of(tableQualifiedName, null));
+            .getAncestor(relationName);
         if (withQuery != null) {
-            relation = withQuery;
+            relationContext.addSourceRelation(withQuery);
+            return withQuery;
         } else {
-            TableInfo tableInfo;
+            String schemaName = relationName.schema();
+            String tableName = relationName.name();
+
+            ArrayList<String> newSearchPath = new ArrayList<>();
+            Iterables.addAll(newSearchPath, searchPath);
+            if (schemaName != null) {
+                newSearchPath.remove(schemaName);
+                newSearchPath.addFirst(schemaName);
+            }
```

Comment:
I'm not sure if this is resulting in the correct behavior. If the user provided a explicit schema name and the table is not found it should always fail. Not continue with the other schema's in the search path.

Why not use the explicit version I posted?